Sources_v3/StreamChatUI/ChatChannel/Attachments/ChatMessageInteractiveAttachmentView.swift

```diff
@@ -0,0 +1,129 @@
+        didSet { updateContentIfNeeded() }
+    }
+
+    // MARK: - Subviews
+
+    public private(set) lazy var preview = uiConfig
+        .messageList
+        .messageContentSubviews
+        .attachmentSubviews
+        .imageGalleryItem
+        .init()
+        .withoutAutoresizingMaskConstraints
+
+    public private(set) lazy var titleLabel: UILabel = {
+        let label = UILabel()
+        label.font = UIFont.preferredFont(forTextStyle: .headline).italic
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .center
+        return label.withoutAutoresizingMaskConstraints
+    }()
+
+    public private(set) lazy var separator = UIView()
+        .withoutAutoresizingMaskConstraints
+
+    public private(set) lazy var actionsStackView: UIStackView = {
+        let stack = UIStackView()
+        stack.axis = .horizontal
+        stack.alignment = .fill
+        stack.distribution = .fillEqually
+        stack.spacing = 1
+        return stack.withoutAutoresizingMaskConstraints
+    }()
+
+    // MARK: - Overrides
+
+    override public func defaultAppearance() {
+        preview.layer.cornerRadius = 8
+        preview.layer.maskedCorners = [.layerMinXMinYCorner, .layerMaxXMinYCorner]
+        preview.clipsToBounds = true
+
+        separator.backgroundColor = uiConfig.colorPalette.messageInteractiveAttachmentActionsBorder
+    }
+
+    override open func setUpLayout() {
+        addSubview(preview)
+        addSubview(titleLabel)
+        addSubview(separator)
+        addSubview(actionsStackView)
+
+        NSLayoutConstraint.activate([
+            preview.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor),
+            preview.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor),
+            preview.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+            preview.heightAnchor.constraint(equalTo: preview.widthAnchor),
+            
+            titleLabel.topAnchor.constraint(equalToSystemSpacingBelow: preview.bottomAnchor, multiplier: 1),
+            titleLabel.leadingAnchor.constraint(equalTo: leadingAnchor),
+            titleLabel.trailingAnchor.constraint(equalTo: trailingAnchor),
+            
+            separator.topAnchor.constraint(equalToSystemSpacingBelow: titleLabel.bottomAnchor, multiplier: 1),
+            separator.leadingAnchor.constraint(equalTo: leadingAnchor),
+            separator.trailingAnchor.constraint(equalTo: trailingAnchor),
+            separator.heightAnchor.constraint(equalToConstant: 1),
+            
+            actionsStackView.topAnchor.constraint(equalTo: separator.bottomAnchor),
+            actionsStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            actionsStackView.trailingAnchor.constraint(equalTo: trailingAnchor),
+            actionsStackView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            actionsStackView.heightAnchor.constraint(equalToConstant: 48)
```

Comment:
😱

Reply:
Introduced `ActionButton` and made height configurable via `defaultAppearance` in 78de8d4f4fe108f6053d83f1793e9f604ef090b2

Tried to go with content based height - buttons are too small. We need some explicitly provided inset or height for better UX (preferred height since inset approach decreases button tap area)